File: sdks/ruby/lib/novabilling/types/api_key_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class APIKeyResponse < Internal::Types::Model
      field :id, -> { String }, optional: false, nullable: false
      field :key, -> { String }, optional: false, nullable: false
      field :name, -> { String }, optional: false, nullable: false
      field :scopes, -> { Internal::Types::Array[String] }, optional: false, nullable: false
      field :last_used, -> { String }, optional: true, nullable: false, api_name: "lastUsed"
      field :expires_at, -> { String }, optional: true, nullable: false, api_name: "expiresAt"
      field :created_at, -> { String }, optional: false, nullable: false, api_name: "createdAt"
    end
  end
end

File: sdks/csharp/src/NovaBilling/Types/ApiKeyResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ApiKeyResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("id")]
    public required string Id { get; set; }

    [JsonPropertyName("key")]
    public required string Key { get; set; }

    [JsonPropertyName("name")]
    public required string Name { get; set; }

    [JsonPropertyName("scopes")]
    public IEnumerable<string> Scopes { get; set; } = new List<string>();

    [JsonPropertyName("lastUsed")]
    public string? LastUsed { get; set; }

    [JsonPropertyName("expiresAt")]
    public string? ExpiresAt { get; set; }

    [JsonPropertyName("createdAt")]
    public required string CreatedAt { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
